src/company_gui.cpp

```diff
@@ -2650,8 +2650,8 @@ struct CompanyWindow : Window
 			default: NOT_REACHED();
 
 			case WID_C_GIVE_MONEY: {
-				Money money = (Money)(std::strtoull(str, nullptr, 10) / _currency->rate);
-				uint32 money_c = Clamp(ClampToI32(money), 0, 20000000); // Clamp between 20 million and 0
+				uint32_t money = ClampTo<uint32_t>(std::strtoull(str, nullptr, 10) / _currency->rate);
+				uint32 money_c = std::min(money, 20000000U); // Maximum is 20 million
 
 				Command<CMD_GIVE_MONEY>::Post(STR_ERROR_CAN_T_GIVE_MONEY, money_c, (CompanyID)this->window_number);
```

Comment:
Probably a bit too much patch-creep, but can we get rid of `uint32 money_c` here and just keep it as `money`?

Reply:
I think so... but we'll have to wait on the compilers for the final verdict.